### DISCUSSION TITLE
Little lolMiner front-end

### DISCUSSION THREAD

**Discussion by AzidHouse** [Lolliedieb/lolMiner-releases#312] [Show and tell]
Hi everybody,

I've developed a little front-end for me to keep track of mining performance at a glance.

This little piece of code listen changes on log file an try to represent data in a visual manner.

BTW, I'm asking the developer of lolMiner if it could be possible to have timestamps on every log line. I've found that it's a little difficult to group data into consistent timeframes (eg.: how much shares on a 10 or 30 minutes lapse) to be presented as a live and historic chart (I re-read the whole old log each time the app starts to reconstruct the data).

I know that there's an API mode to get insights, but I've found It's a little too basic and can offer only real-time stats, so I've opted for the old good log file hook.

If someone is interested I could open a github project, later on.

In this screenshot you'll see that I have info only for a single GPU (I've got only one, sorry), I will enhance it to show in a tab panel with every single GPU stats and a general summary.

Any idea is welcome.

![image](https://user-images.githubusercontent.com/10941806/109060396-a7383880-76e5-11eb-9afc-9ffaff519f6c.png)

**Comment by ChesDonno** [comment]
nice!

**Comment by mew905** [comment]
thats a great start! now if you could add config options to make it a true front-end (rather than just a monitor) I would definitely be interested in giving this a go

**Comment by AzidHouse** [comment]
Thank you, I'm just adding the option to configure and run/stop lolMiner directly form the gui.
Further improvements (feel free to add your own suggestions):
- Multiple GPU Support (obviously)
- Alarms, also via email (under-performance, stale shares, fan/power/temp, gpu down, etc.)
- Hystorical data retention & search (sessions, performance, etc.)
- Mining Pool API connector (to show actual pool data)
- Ethereum network support (balance, actual transactions, etc.)?

BTW, I'm not sure if this should be a desktop app, in the end. Maybe should be better a web app to be able to access the interface from everywhere. By now I'm trying to complete the core funtionalities, later on we'll see.
